community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/BadCollector.java

```diff
@@ -56,16 +68,44 @@ public BadCollector( OutputStream out, int tolerance, int collect )
     }
 
     @Override
-    public void collectBadRelationship( InputRelationship relationship, Object specificValue )
+    public void collectBadRelationship( final InputRelationship relationship, final Object specificValue )
     {
-        checkTolerance( BAD_RELATIONSHIPS,
-                new InputException( format( "%s refering to missing node %s", relationship, specificValue ) ) );
+        checkTolerance( BAD_RELATIONSHIPS, new Bad()
```

Comment:
The whole `Bad` and everything it does is short lived. Those instances will only make it into that report method, log a message and potentially throw an exception before dying.